homeassistant/components/google_photos/api.py

```diff
@@ -61,14 +64,38 @@ async def get_media_item(self, media_item_id: str) -> dict[str, Any]:
         return await self._execute(cmd)
 
     async def list_media_items(
-        self, page_size: int | None = None, page_token: str | None = None
+        self,
+        page_size: int | None = None,
+        page_token: str | None = None,
+        album_id: str | None = None,
+        favorites: bool = False,
     ) -> dict[str, Any]:
         """Get all MediaItem resources."""
         service = await self._get_photos_service()
-        cmd: HttpRequest = service.mediaItems().list(
+        args: dict[str, Any] = {
+            "pageSize": (page_size or DEFAULT_PAGE_SIZE),
+            "pageToken": page_token,
+        }
+        cmd: HttpRequest
+        if album_id is not None or favorites:
+            if album_id is not None:
+                args["albumId"] = album_id
+            if favorites:
+                args["filters"] = {"featureFilter": {"includedFeatures": "FAVORITES"}}
```

Comment:
From the API docs: "Filters to apply to the request. Can't be set in conjunction with an albumId."

Reply:
I saw this. I figured that it would be reasonable to let the API throw an error. I didn't want additional untested lines given that there isn't code we can exercise to test both.  I could add this validation when this is moved to an external library, howerver...